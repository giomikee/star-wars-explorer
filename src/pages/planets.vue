<template>
  <v-container>
    <h2>Planets Database</h2>
  </v-container>

  <SearchNameInput
    :is-loading="arePlanetsEmpty"
    @name-searched="setSearchedName"
  />

  <SortButtons
    :is-loading="arePlanetsEmpty"
    @sorted:date="sortPlanetsByDate"
    @sorted:name="sortPlanetsByName"
  />

  <TableWrapper
    :current-page="currentPage"
    :is-loading="arePlanetsEmpty"
    :pages="itemsPages"
    @page-changed="updateCurrentPage"
  >
    <PlanetsTable :current-page="currentPage" />
  </TableWrapper>
</template>

<script setup lang="ts">
  import PlanetsTable from '../components/PlanetsTable/PlanetsTable.vue'
  import SearchNameInput from '../components/SearchNameInput/SearchNameInput.vue'
  import TableWrapper from '../components/TableWrapper/TableWrapper.vue'
  import { usePlanetsPage } from './composables'

  const {
    arePlanetsEmpty,
    currentPage,
    itemsPages,
    setSearchedName,
    sortPlanetsByDate,
    sortPlanetsByName,
    updateCurrentPage,
  } = usePlanetsPage()
</script>
