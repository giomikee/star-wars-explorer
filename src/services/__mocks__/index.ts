export * from './people.service'
export * from './planets.service'
