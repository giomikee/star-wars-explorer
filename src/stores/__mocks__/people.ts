import type { People } from '../../interfaces'

export const mockedPeople: People[] = [
  {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person2',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person3',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person4',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person5',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person6',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person7',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person8',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person9',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person10',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person11',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
  {
    name: 'Person12',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.info/api/planets/1',
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/2',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/6',
    ],
    species: [],
    vehicles: [
      'https://swapi.info/api/vehicles/14',
      'https://swapi.info/api/vehicles/30',
    ],
    starships: [
      'https://swapi.info/api/starships/12',
      'https://swapi.info/api/starships/22',
    ],
    created: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.info/api/people/1',
  },
]
