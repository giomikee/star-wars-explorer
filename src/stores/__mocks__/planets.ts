import type { Planet } from '../../interfaces'

const mockedDate = new Date('2014-12-09T13:50:49.641000Z')

export const mockedPlanets: Planet[] = [
  {
    name: 'Tatooine',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet2',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet3',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet4',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet5',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet6',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet7',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet8',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet9',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet10',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet11',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
  {
    name: 'Planet12',
    rotation_period: '23',
    orbital_period: '304',
    diameter: '10465',
    climate: 'arid',
    gravity: '1 standard',
    terrain: 'desert',
    surface_water: '1',
    population: '200000',
    residents: [
      'https://swapi.info/api/people/1',
      'https://swapi.info/api/people/2',
      'https://swapi.info/api/people/4',
      'https://swapi.info/api/people/6',
      'https://swapi.info/api/people/7',
      'https://swapi.info/api/people/8',
      'https://swapi.info/api/people/9',
      'https://swapi.info/api/people/11',
      'https://swapi.info/api/people/43',
      'https://swapi.info/api/people/62',
    ],
    films: [
      'https://swapi.info/api/films/1',
      'https://swapi.info/api/films/3',
      'https://swapi.info/api/films/4',
      'https://swapi.info/api/films/5',
      'https://swapi.info/api/films/6',
    ],
    created: '2014-12-09T13:50:49.641000Z',
    edited: mockedDate,
    url: 'https://swapi.info/api/planets/1',
  },
]
