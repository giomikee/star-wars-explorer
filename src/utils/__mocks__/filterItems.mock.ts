import type { Item } from '../../interfaces'

export const mockedDate = new Date('2014-12-09T13:50:49.641000Z')

export const mockedItems: Item[] = [
  {
    name: 'Foo',
    created: mockedDate,
  },
  {
    name: 'Food',
    created: mockedDate,
  },
  {
    name: 'Bar',
    created: mockedDate,
  },
]
