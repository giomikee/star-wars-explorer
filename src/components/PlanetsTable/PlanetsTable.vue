<template>
  <v-table class="w-100">
    <thead>
      <tr>
        <th scope="col">Name</th>
        <th scope="col">Rotation Period</th>
        <th scope="col">Orbital Period</th>
        <th scope="col">Diameter</th>
        <th scope="col">Climate</th>
        <th scope="col">Gravity</th>
        <th scope="col">Terrain</th>
        <th scope="col">Surface water</th>
        <th scope="col">Population</th>
        <th scope="col">Date Added</th>
      </tr>
    </thead>
    <tbody>
      <tr v-for="planet in paginatedPlanets" :key="planet.url">
        <td>{{ planet.name }}</td>
        <td>{{ planet.rotation_period }}</td>
        <td> {{ planet.orbital_period }}</td>
        <td>{{ planet.diameter }}</td>
        <td>{{ planet.climate }}</td>
        <td>{{ planet.gravity }}</td>
        <td>{{ planet.terrain }}</td>
        <td>{{ planet.surface_water }}</td>
        <td>{{ planet.population }}</td>
        <td>{{ formatDateTime(planet.created) }}</td>
      </tr>
    </tbody>
  </v-table>
</template>

<script lang="ts" setup>
  import type { Planet } from '../../interfaces'
  import { computed } from 'vue'
  import { usePlanetsStore } from '../../stores/planets.store'
  import { formatDateTime, getPaginatedItems } from '../../utils'

  interface Props {
    currentPage: number
  }

  const props = defineProps<Props>()
  const { getPlanetsToDisplay } = usePlanetsStore()

  const paginatedPlanets = computed<Planet[]>(() => getPaginatedItems(getPlanetsToDisplay(), props.currentPage))
</script>

<script lang="ts">
  export default { name: 'PlanetTable' }
</script>
