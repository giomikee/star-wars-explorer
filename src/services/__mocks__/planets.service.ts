import { mockedPlanets } from '../../stores/__mocks__/planets'

export function getAllPlanets () {
  return mockedPlanets
}
