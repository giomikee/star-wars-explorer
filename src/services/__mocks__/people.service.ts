import { mockedPeople } from '../../stores/__mocks__/people'

export function getAllPeople () {
  return mockedPeople
}
