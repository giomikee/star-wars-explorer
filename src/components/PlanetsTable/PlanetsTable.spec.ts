import type { Planet } from '../../interfaces'
import { shallowMount } from '@vue/test-utils'
import { beforeAll, describe, expect, it, vi } from 'vitest'
import { mockedPlanets } from '../../stores/__mocks__/planets'
import { usePlanetsStore } from '../../stores/planets.store'
import PlanetsTable from './PlanetsTable.vue'

const defaultProps = { currentPage: 1 }

interface Component {
  paginatedPlanets: Planet[]
}

const createWrapper = (props = defaultProps) => shallowMount(
  PlanetsTable,
  {
    props,
    global: {
      stubs: {
        VTable: false,
      },
    },
  })

describe('Given PlanetsTable component', () => {
  beforeAll(() => {
    const store = usePlanetsStore()
    vi.spyOn(store, 'getPlanetsToDisplay').mockReturnValue(mockedPlanets)
  })

  describe('When component is rendered', () => {
    it('Then it should match snapshot', () => {
      const { element } = createWrapper()

      expect(element).toMatchSnapshot()
    })
  })

  describe('Given computed properties', () => {
    describe.each([
      {
        currentPage: 1,
        expectedLength: 10,
      },
      {
        currentPage: 2,
        expectedLength: 2,
      },
    ])('When paginatedPlanets is evaluated', ({ currentPage, expectedLength }) => {
      describe(`And currentPage prop is ${currentPage}`, () => {
        it(`Then it should return ${expectedLength} Planets from the store`, () => {
          const wrapper = createWrapper({ currentPage })

          expect((wrapper.vm as typeof wrapper.vm & Component).paginatedPlanets).toHaveLength(expectedLength)
        })
      })
    })
  })
})
