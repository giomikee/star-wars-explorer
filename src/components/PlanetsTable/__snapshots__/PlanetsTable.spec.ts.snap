// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given PlanetsTable component > When component is rendered > Then it should match snapshot 1`] = `
<div
  class="v-table v-theme--dark v-table--density-default w-100"
>
  <!---->
  <div
    class="v-table__wrapper"
  >
    <table>
      
      <thead>
        <tr>
          <th
            scope="col"
          >
            Name
          </th>
          <th
            scope="col"
          >
            Rotation Period
          </th>
          <th
            scope="col"
          >
            Orbital Period
          </th>
          <th
            scope="col"
          >
            Diameter
          </th>
          <th
            scope="col"
          >
            Climate
          </th>
          <th
            scope="col"
          >
            Gravity
          </th>
          <th
            scope="col"
          >
            Terrain
          </th>
          <th
            scope="col"
          >
            Surface water
          </th>
          <th
            scope="col"
          >
            Population
          </th>
          <th
            scope="col"
          >
            Date Added
          </th>
        </tr>
      </thead>
      <tbody>
        
        <tr>
          <td>
            Tatooine
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet2
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet3
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet4
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet5
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet6
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet7
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet8
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet9
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        <tr>
          <td>
            Planet10
          </td>
          <td>
            23
          </td>
          <td>
            304
          </td>
          <td>
            10465
          </td>
          <td>
            arid
          </td>
          <td>
            1 standard
          </td>
          <td>
            desert
          </td>
          <td>
            1
          </td>
          <td>
            200000
          </td>
          <td>
            December 9, 2014 at 1:50 PM
          </td>
        </tr>
        
      </tbody>
      
    </table>
  </div>
  <!---->
</div>
`;
